drivers/gpio/gpio_lpc17xx.c

```diff
@@ -0,0 +1,128 @@
+#include <errno.h>
+#include <device.h>
+#include <gpio.h>
+#include <soc.h>
+#include "gpio_utils.h"
+
+#include <sys_io.h>
+
+#define PCONP_REG 0x400FC0C4
+
+enum gpio_regs {
+        GPIO_DIR_OFFSET       = 0x0,
+        GPIO_OUT_SET_OFFSET   = 0x018,
+        GPIO_OUT_CLEAR_OFFSET = 0x01C,
+        GPIO_IN_OFFSET        = 0x014,
+};
+
+#define GPIO_REG_ADDR(base, offset, n) base + offset + (0x20 * n)
+
+#define GET_PORT(pin) (pin / 32) ? 2 :0
+#define GET_PIN(pin) pin % 32
+
+struct gpio_lpc17xx_config {
+        u32_t base;
+};
+
+static int gpio_lpc17xx_configure(struct device *dev,
+                                  int access_op, u32_t pin, int flags)
+{
+        const struct gpio_lpc17xx_config *cfg = dev->config->config_info;
+        u32_t base = cfg->base;
+
+        /* Get Port and Pin number */
+        u32_t port = GET_PORT(pin);
+        pin = GET_PIN(pin);
+
+        /* Check for an invalid pin configuration */
+        if ((flags & GPIO_INT) && (flags & GPIO_DIR_OUT)) {
+                return -EINVAL;
+        }
+        /* Check if GPIO port supports interrupts */
+        if (flags & GPIO_INT) {
+                return -EINVAL;
+        }
+        /* supports access by pin now,you can add access by port when needed */
+        if (access_op == GPIO_ACCESS_BY_PIN) {
```

Comment:
It would be better to write this as:

``````
if (access_op != GPI_ACCESS_BY_PIN) {
    return -EINVAL
}
``````